Sources/Amplitude/Utilities/HttpClient.swift

```diff
@@ -9,13 +9,17 @@ import Foundation
 
 class HttpClient {
     let configuration: Configuration
-    internal var session: URLSession
+    internal let session: URLSession
 
     init(configuration: Configuration) {
         self.configuration = configuration
-        // shared instance has limitations but think we are not affected
-        // https://developer.apple.com/documentation/foundation/urlsession/1409000-shared
-        self.session = URLSession.shared
+        // an exclusive ephemeral session avoids contention (fewer request timeouts)
+        let sessionConfiguration = URLSessionConfiguration.ephemeral
+        // add some throughput
+        sessionConfiguration.httpMaximumConnectionsPerHost = 2
+        // avoid keeping app background task alive for too long
+        sessionConfiguration.timeoutIntervalForRequest = 20
```

Comment:
This will limit all requests to 20s, vs default of 60s which could increase timeouts in low latency situations. I would prefer to leave this as the default.

Also should we use [timeoutIntervalForResource](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1408153-timeoutintervalforresource) be used instead of [timeoutIntervalForRequest](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1408259-timeoutintervalforrequest)?

Reply:
[timeoutIntervalForRequest](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1408259-timeoutintervalforrequest)? is the maximum duration a task should wait for additional data to arrive before giving up. In other words, it will cancel a partially uploaded task if it's been greater than the timeout interval since the last packet was sent. The timer is resent each time the task progresses.

the default [timeoutIntervalForResource](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/1408153-timeoutintervalforresource) is 7 days, and is the maximum duration a task should take from initiation to completion. It's more relevant for long running downloads or uploads. I think the default is fine here. 

In terms of whether it should be 20s or 60s, 20s has the benefit of being under the background task deadline time (30s) so helps avoids the app being penalised by the system for running for too long. Also, as flush() is automatically being called every 30s or so, any timeout task will soon be retried. This isn't a huge deal though – I'm happy with 60s if you're more comfortable with it.